test/FunctionalTest.java

```diff
@@ -1,89 +1,113 @@
-import org.junit.*;
+import com.google.inject.*;
 
 import java.util.*;
 
+import controllers.routes;
+import play.*;
+import play.inject.guice.*;
 import play.mvc.*;
 
-import static play.test.Helpers.*;
-import static org.fest.assertions.Assertions.*;
+import play.test.*;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+import org.junit.*;
+
+import javax.inject.Inject;
 
 public class FunctionalTest {
 
+    @Inject
+    Application application;
+
+    @Before
+    public void setup() {
+        Module testModule = new AbstractModule() {
+            @Override
+            public void configure() {
+                // Install custom test binding here
+            }
+        };
+
+        GuiceApplicationBuilder builder = new GuiceApplicationLoader()
+                .builder(new ApplicationLoader.Context(Environment.simple()))
+                .overrides(testModule);
+        Guice.createInjector(builder.applicationModule()).injectMembers(this);
+
+        Helpers.start(application);
+    }
+
+
+    @After
+    public void teardown() {
+        Helpers.stop(application);
+    }
+
     @Test
     public void redirectHomePage() {
-        running(fakeApplication(), new Runnable() {
-           public void run() {
-               Result result = callAction(controllers.routes.ref.Application.index());
+        Helpers.running(Helpers.fakeApplication(), () -> {
```

Comment:
You won't need both `application` built at the `setup` method and this `fakeApplication` here. In other words, all these `Helpers.running` could be removed and the test be "plain".